openstates/oh/bills.py

```diff
@@ -13,43 +13,333 @@
+                            #so we'll check and throw an error if we find one
+                            #life is fragile. so are our scrapers.
+                            if "veto" in bill_version:
+                                veto_url = base_url+bill_version["veto"][0]["link"]
+                                veto_json = self.get(veto_url).json()
+                                if len(veto_json["items"]) > 0:
+                                    raise AssertionError("Whoa, a veto! We've never gotten one before. Go write some code to deal with it: {}".format(veto_url))
+
+                            if "disapprove" in bill_version:
+                                disapprove_url = base_url+bill_version["disapprove"][0]["link"]
+                                disapprove_json = self.get(disapprove_url).json()
+                                if len(disapprove_json["items"]) > 0:
+                                    raise AssertionError("Whoa, a disapprove! We've never gotten one before. Go write some code to deal with it: {}".format(disapprove_url))
+
+
+
+
+                        #this stuff is version-specific
+                        version_name = bill_version["version"]
+                        version_link = base_url+bill_version["pdfDownloadLink"]
+                        mimetype = "application/pdf" if version_link.endswith("pdf") else None
+                        bill.add_version(version_name,version_link,mimetype=mimetype)
+
+                    self.save_bill(bill)
+
+    def pages(self,base_url, first_page):
+        page = self.get(first_page)
+        page = page.json()
+        yield page
+        while "nextLink" in page:
+            page = self.get(base_url+page["nextLink"])
+            page = page.json()
+            yield page
+
+
+    def get_other_data_source(self,first_page,base_url,source_name):
+        #produces a dictionary from bill_id to a list of
+        #one of the following:
+        #amendments, analysis, fiscals, synopsis
+        #could pull these by bill, but doing it in bulk
+        #and then matching on our end will get us by with way fewer
+        #api calls
+        
+        bill_dict = {}
+        for page in self.pages(base_url,first_page+source_name):
+            for item in page["items"]:
+                billno = item["billno"]
+                if billno not in bill_dict:
+                    bill_dict[billno] = []
+                bill_dict[billno].append(item)
+
+        return bill_dict
+
+
+
+    def add_document(self,documents,bill_id,type_of_document,bill,first_page):
+        try:
+            documents = documents[bill_id]
+        except KeyError:
+            return
+        for item in documents:
+            if type_of_document == "amendment":
+                name = item["amendnum"] + " " + item["version"]
+            else:
+                name = item["name"] or type_of_document
+            link = first_page+item["link"]+"?format=pdf"
```

Comment:
You might want to add a check to make sure the PDF version exists, although from what I've seen it does in all cases.